File: m26-js/test/m26_distance_spec.js
(function() {
  describe('Distance', function() {
    it("should assume miles as UOM, and convert to other units", function() {
      var d;
      d = new Distance(26.2);
      expect(d.uom()).toBe(Constants.UOM_MILES);
      expect(d.as_miles()).isWithin(0.0000000001, 26.2);
      expect(d.as_kilometers()).isWithin(0.0000000001, 42.1648128);
      return expect(d.as_yards()).isWithin(0.000001, 46112.0);
    });
    it("should calculate a 10K, and convert to other units", function() {
      var d;
      d = new Distance(10.0, 'k');
      expect(d.uom()).toBe(Constants.UOM_KILOMETERS);
      expect(d.as_miles()).isWithin(0.0000000001, 6.2137119223733395);
      expect(d.as_kilometers()).isWithin(0.0000000001, 10.0);
      return expect(d.as_yards()).isWithin(0.000001, 10936.132983377078);
    });
    it("should calculate an 1800y, and convert to other units", function() {
      var d;
      d = new Distance(1800.0, 'y');
      expect(d.uom()).toBe(Constants.UOM_YARDS);
      expect(d.as_miles()).isWithin(0.0000000001, 1.0227272727272727);
      expect(d.as_kilometers()).isWithin(0.0000000001, 1.64592);
      return expect(d.as_yards()).isWithin(0.000001, 1800);
    });
    it("should add", function() {
      var d1, d2, d3;
      d1 = new Distance(26.2);
      d2 = new Distance(4.8);
      d3 = d1.add(d2);
      expect(d3.uom()).toBe(Constants.UOM_MILES);
      return expect(d3.as_miles()).isWithin(0.0000000001, 31.0);
    });
    return it("should subtract", function() {
      var d1, d2, d3;
      d1 = new Distance(26.2);
      d2 = new Distance(10.0, 'k');
      d3 = d1.subtract(d2);
      expect(d3.uom()).toBe(Constants.UOM_MILES);
      return expect(d3.as_miles()).isWithin(0.0000000001, 19.98628807762666);
    });
  });

}).call(this);

File: m26/spec/m26_distance_spec.rb
=begin

Copyright (C) 2014 Chris Joakim, JoakimSoftware LLC

=end

require 'spec_helper'

describe "Test class M26::Distance" do

  before :all do
  end

  before :each do  # :each is the default 'before' value
  end

  after :each do
  end

  after :all do
  end

  it "should assume miles as UOM" do
    @d = M26::Distance.new(26.2)
    @d.should be_valid
    @d.should_not be_an_instance_of String
    @d.should be_an_instance_of M26::Distance
    @d.uom.should eql('m')
  end

  it "should return its value" do
    @d = M26::Distance.new(26.2)
    @d.value.should == 26.2
  end

  it "should convert from miles" do
    @d = M26::Distance.new(6.20)
    @d.should be_valid
    @d.get_miles.should be_within(0.001).of(6.2)
    @d.get_kilometers.should be_within(0.001).of(10.0)
    @d.get_yards.should be_within(0.001).of(10912.0)

    @d = M26::Distance.new(6.20, 'm')
    @d.should be_valid
    @d.get_miles.should be_within(0.001).of(6.2)
    @d.get_kilometers.should be_within(0.001).of(10.0)
    @d.get_yards.should be_within(0.001).of(10912.0)
  end

  it "should convert from kilometers" do
    @d = M26::Distance.new(10.0, 'k')
    @d.should be_valid
    @d.get_miles.should be_within(0.001).of(6.2)
    @d.get_kilometers.should be_within(0.001).of(10.0)
    @d.get_yards.should be_within(0.001).of(10912.0)
  end

  it "should convert from yards" do
    @d = M26::Distance.new(10912.0, 'y')
    @d.should be_valid
    @d.get_miles.should be_within(0.001).of(6.2)
    @d.get_kilometers.should be_within(0.001).of(10.0)
    @d.get_yards.should be_within(0.001).of(10912.0)
  end

  it "should not recognize unknown units of distance" do
    @d = M26::Distance.new(444.44, 'light_years')
    @d.should_not be_valid
  end

  it "should subtract" do
    @d1 = M26::Distance.new(18.0)
    @d2 = M26::Distance.new(26.2)
    @d2.instance_of?(M26::Distance).should == true
    @d2.subtract(@d1).get_miles.should == 8.2
  end

  it "should to_s" do
    @d = M26::Distance.new(26.2)
    @d.to_s.should == 'Distance: 26.2 m'
  end

  it "should print_string" do
    @d = M26::Distance.new(26.2)
    @d.print_string.should == 'Distance: 26.2 m 26.2 42.258064516222 46112.0'
  end

end
